packages/block-library/src/image/edit.js

```diff
@@ -425,10 +406,25 @@ class ImageEdit extends Component {
 				<Fragment>
 					{ controls }
 					<MediaPlaceholder
-						icon={ <BlockIcon icon={ icon } /> }
-						className={ className }
+						icon={ ! url ? <BlockIcon icon={ icon } /> : <img
+							alt={ __( 'Replace image' ) }
+							title={ __( 'Replace image' ) }
+							width={ '50%' }
+							className={ 'replace-image-preview' }
+							src={ url }
+						/> }
+						labels={ {
+							title: ! url ? __( 'Image' ) : <span
+								className={ 'replace-image-preview-title' }
```

Comment:
We use something approximating BEM for our class names. Maybe `replace-image-preview__title`?

https://wordpress.org/gutenberg/handbook/contributors/develop/coding-guidelines/#naming